zebra/zebra_nhg.c

```diff
@@ -3526,7 +3526,11 @@ void zebra_nhg_uninstall_kernel(struct nhg_hash_entry *nhe)
 		}
 	}
 
-	zebra_nhg_handle_uninstall(nhe);
+	/* do not call delete of the NHG if there are re's referencing it. */
+	if (free_nhg)
```

Comment:
so ... the logic here just seems incorrect: this code places the decision-making for the fate of the nhg object entirely outside that object?